languages/csharp-dotnet-2.2/template/pipe/Module.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using Newtonsoft.Json;
+using System.Reflection;
+using __ALGO__;
+
+namespace __ALGO__.Pipe
+{
+    public class Module
+    {
+
+        private Type _primaryClassType;
+        private Type _algoInputType;
+        private MethodInfo _applyMethod;
+        
+        
+        public Module()
+        {
+            _primaryClassType = typeof(__ALGO__);
+            GetApplyMethod();
+            GetAlgoInputType();
+        }
+        
+
+        private void GetApplyMethod()
+        {
+            // getting only public methods, as those are apply methods
+            MethodInfo applyMethod = _primaryClassType.GetMethod("apply");
```

Comment:
We're only taking the first apply, in most instances this will have zero impact on functionality.